test/common/config/registry_test.cc

```diff
@@ -182,6 +195,50 @@ TEST(RegistryTest, TestDoubleRegistrationByName) {
                             "Double registration for name: 'testing.published.test'");
 }
 
+class TestVersionedWithDeprecatedNamesFactoryAndAdditionalCategory : public PublishedFactory {
```

Comment:
I think that the deprecated name comes into play on the call to register.

A better name for this class would be: PublishedFactoryWithNameAndCategory